video/out/hwdec/hwdec_vaapi.c

```diff
@@ -301,7 +302,7 @@ static int mapper_map(struct ra_hwdec_mapper *mapper)
     // We can handle composed formats if the total number of planes is still
     // equal the number of planes we expect. Complex formats with auxilliary
     // planes cannot be supported.
-    if (p->num_planes != num_returned_planes) {
+    if (p->num_planes != 0 && p->num_planes != num_returned_planes) {
```

Comment:
Why the change here? Because you noticed `GRAY8` has zero planes? I don't know what's up with that, but I don't think it's wrong to reject a malformed format like that.

Reply:
This was added to differentiate `dmabuf _interop_wl` from others, as this interop doesn't need to set the number of planes. In this case, we accept formats based purely on what Wayland reports it supports, so we ignore this check on returned planes.